src/app/public/modules/data-manager/data-manager.service.spec.ts

```diff
@@ -202,38 +202,125 @@ describe('SkyDataManagerService', () => {
+          dataState = currentDataState;
+
+          dataStateObservable.subscribe(state => {
+            dataState = state;
+          });
+        });
 
-        dataStateObservable.subscribe(state => {
-          dataState = state;
+        it('should emit new values when the sourceId of the change is different from the given sourceId', () => {
+          const startingDataState = dataState;
+
+          expect(dataState).toEqual(startingDataState);
+
+          dataManagerService.updateDataState(dataStateUpdate, otherSourceId);
+
+          expect(dataState).not.toEqual(startingDataState);
+          expect(dataState).toEqual(dataStateUpdate);
+        });
+
+        it('should not emit new values when the sourceId of the change matches the given sourceId', () => {
+          const startingDataState = dataState;
+
+          expect(dataState).toEqual(startingDataState);
+
+          dataManagerService.updateDataState(dataStateUpdate, sourceId);
+
+          expect(dataState).toEqual(startingDataState);
         });
       });
 
-      it('should emit new values when the sourceId of the change is different from the given sourceId', () => {
-        const startingDataState = dataState;
+      describe('with filters', () => {
+        const filterProperties = ['searchText', 'selectedIds'];
+
+        beforeEach(() => {
+          dataStateObservable = dataManagerService.getDataStateUpdates(sourceId, { properties: filterProperties });
+          dataState = currentDataState;
+
+          dataStateObservable.subscribe(state => {
+            dataState = state;
+          });
+        });
+
+        it('should emit new values when one of the given filter properties changes', () => {
+          const startingDataState = dataState;
+
+          expect(dataState).toEqual(startingDataState);
+
+          dataManagerService.updateDataState(dataStateUpdate, otherSourceId);
+
+          expect(dataState).not.toEqual(startingDataState);
+          expect(dataState).toEqual(dataStateUpdate);
+        });
+
+        it('should not emit new values if a property that is not a given filter property changes', () => {
+          const startingDataState = dataState;
+          const newDataState = dataState;
 
-        expect(dataState).toEqual(startingDataState);
+          expect(dataState).toEqual(startingDataState);
 
-        dataManagerService.updateDataState(dataStateUpdate, 'differentId');
+          newDataState.onlyShowSelected = !newDataState.onlyShowSelected;
 
-        expect(dataState).not.toEqual(startingDataState);
-        expect(dataState).toEqual(dataStateUpdate);
+          dataManagerService.updateDataState(newDataState, otherSourceId);
+
+          expect(dataState).toEqual(startingDataState);
+        });
       });
+      describe('with comparator', () => {
+        const firstDataState = new SkyDataManagerState({ searchText: 'SEARCHTEXT' });
+        console.log('first data state');
```

Comment:
Are the `console.log` lines left over from debugging? Can they be removed?

Reply:
lol woops, completely forgot to remove those